File: delivery_agency/src/main/java/dto/ClientDto.java
package dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Component;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Component
public class ClientDto {
    private Long id;
    @NotBlank(message = "Client name cannot be empty")
    @Size(min = 1, max = 50, message = "Client name size must be between 1 and 50 characters")
    private String name;
    @NotBlank(message = "Client surname cannot be empty")
    @Size(min = 1, max = 50, message = "Client surname size must be between 1 and 50 characters")
    private String surname;
    @NotBlank(message = "Client passportId cannot be empty")
    @Size(min = 1, max = 100, message = "Client passportId must be between 1 and 100 characters")
    private String passportId;
    @NotBlank(message = "Client phoneNumber cannot be empty")
    @Size(min = 1, max = 100, message = "Client phoneNumber must be between 1 and 100 characters")
    private String phoneNumber;
    @NotEmpty(message = "Client must have at least one order")
    private List<OrderDto> orders;
}

File: delivery_agency/src/main/resources/static/js/dto/ClientDto.js
export class ClientDto {
    constructor(props = {}) {
        this._id = props.id;
        this._name = props.name;
        this._surname = props.surname;
        this._passportId = props.passportId;
        this._phoneNumber = props.phoneNumber;
        this._orders = props.orders;
    }

    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get name() {
        return this._name;
    }

    set name(value) {
        this._name = value;
    }

    get surname() {
        return this._surname;
    }

    set surname(value) {
        this._surname = value;
    }

    get passportId() {
        return this._passportId;
    }

    set passportId(value) {
        this._passportId = value;
    }

    get phoneNumber() {
        return this._phoneNumber;
    }

    set phoneNumber(value) {
        this._phoneNumber = value;
    }

    get orders() {
        return this._orders;
    }

    set orders(value) {
        this._orders = value;
    }

    toJSON(){
        return {
            id: this._id,
            name: this._name,
            surname: this._surname,
            passportId: this._passportId,
            phoneNumber: this._phoneNumber,
            orders: this._orders
        };
    }
}
